src/backend/distributed/executor/multi_utility.c

```diff
@@ -1987,15 +1984,49 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 	}
 
 	EnsureCoordinator();
-	ShowNoticeIfNotUsing2PC();
 
-	if (shouldSyncMetadata)
+	if (!ddlJob->concurrent)
 	{
-		SendCommandToWorkers(WORKERS_WITH_METADATA, DISABLE_DDL_PROPAGATION);
-		SendCommandToWorkers(WORKERS_WITH_METADATA, (char *) ddlJob->commandString);
+		ShowNoticeIfNotUsing2PC();
+
+		if (shouldSyncMetadata)
+		{
+			SendCommandToWorkers(WORKERS_WITH_METADATA, DISABLE_DDL_PROPAGATION);
+			SendCommandToWorkers(WORKERS_WITH_METADATA, (char *) ddlJob->commandString);
+		}
+
+		ExecuteModifyTasksWithoutResults(ddlJob->taskList);
 	}
+	else
+	{
+		/* save old commit protocol to restore at xact end */
+		Assert(SavedMultiShardCommitProtocol == COMMIT_PROTOCOL_BARE);
+		SavedMultiShardCommitProtocol = MultiShardCommitProtocol;
+		MultiShardCommitProtocol = COMMIT_PROTOCOL_BARE;
+
+		PG_TRY();
+		{
+			if (shouldSyncMetadata)
```

Comment:
Sure, I'll swap them. Ultimately, the coordinator node will be marked `INVALID`, which close to as good as we can do, without a bunch of custom hackery.